Library/Homebrew/dev-cmd/contributions.rb

```diff
@@ -0,0 +1,102 @@
+
+  sig { returns(CLI::Parser) }
+  def contributions_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `contributions`
+
+        Contributions to Homebrew repos for a user.
+      EOS
+
+      flag "--email=",
+           description: "A user's email address that they commit with."
+
+      flag "--from=",
+           description: "Date (ISO-8601 format) to start searching contributions."
+
+      flag "--to=",
+           description: "Date (ISO-8601 format) to stop searching contributions."
+
+      comma_array "--repos=",
+                  description: "The Homebrew repositories to search for contributions in. " \
+                               "Comma separated. Supported repos: #{SUPPORTED_REPOS.join(", ")}."
+
+      named_args :none
+    end
+  end
+
+  sig { returns(NilClass) }
+  def contributions
+    args = contributions_args.parse
+
+    return ofail "`--repos` and `--email` are required." if !args[:repos] || !args[:email]
+
+    commits = 0
+    coauthorships = 0
+
+    args[:repos].each do |repo|
+      if SUPPORTED_REPOS.exclude?(repo)
+        return ofail "Unsupported repo: #{repo}. Try one of #{SUPPORTED_REPOS.join(", ")}."
+      end
+
+      repo_path = find_repo_path_for_repo(repo)
+      return ofail "Couldn't find repo #{repo} locally. Do you have it tapped?" unless Dir.exist?(repo_path)
+
+      commits += git_log_cmd("author", repo_path, args)
+      coauthorships += git_log_cmd("coauthorships", repo_path, args)
+    end
+
+    sentence = "Person #{args[:email]} directly authored #{commits} commits"
+    sentence += " and co-authored #{coauthorships} commits"
+    sentence += " to #{args[:repos].join(", ")}"
+    sentence += if args[:from] && args[:to]
+      " between #{args[:from]} and #{args[:to]}"
+    elsif args[:from]
+      " after #{args[:from]}"
+    elsif args[:to]
+      " before #{args[:to]}"
+    else
+      " in all time"
+    end
+    sentence += "."
+
+    puts sentence
+  end
+
+  sig { params(repo: String).returns(T.nilable(String)) }
+  def find_repo_path_for_repo(repo)
+    case repo
+    when "brew"
+      HOMEBREW_REPOSITORY
+    when "core"
+      "#{HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-core"
+    when "cask"
+      "#{HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-cask"
+    when "bundle"
+      "#{HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-bundle"
+    end
+  end
+
+  sig { params(kind: String, repo_path: String, args: Homebrew::CLI::Args).returns(Integer) }
+  def git_log_cmd(kind, repo_path, args)
+    cmd = "git -C #{repo_path} log --oneline"
+    cmd += " --author=#{args[:email]}" if kind == "author"
+    cmd += " --format='%(trailers:key=Co-authored-by:)'" if kind == "coauthorships"
+    cmd += " --before=#{args[:to]}" if args[:to]
+    cmd += " --after=#{args[:from]}" if args[:from]
+    cmd += " | grep #{args[:email]}" if kind == "coauthorships"
+
+    `#{cmd} | wc -l`.strip.to_i
+  end
```

Comment:
The logic here seems different enough compared to the size of the method that I wonder if there should be separate methods for authors and coauthorships